src/Backends/InterfaceSettings.vala

```diff
@@ -57,5 +78,44 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
         accounts_service.cursor_size = interface_settings.get_int (CURSOR_SIZE);
         accounts_service.locate_pointer = interface_settings.get_boolean (LOCATE_POINTER);
         accounts_service.text_scaling_factor = interface_settings.get_double (TEXT_SCALING_FACTOR);
+
+        accounts_service.picture_options = background_settings.get_enum (PICTURE_OPTIONS);
+        accounts_service.primary_color = background_settings.get_string (PRIMARY_COLOR);
+    }
+
+    private void sync_background_to_greeter () {
+        // File.new_for_uri creates file with broken get_basename method, so do this
```

Comment:
Can you get more in details about this? `File.new_for_uri()` here get a correct basename, only if this is not a uri, then it returns nothing. what's expected.

settings values can be always be set to invalid values by external changes. so checking that the file exists should be done. if the `picture-uri` value points to nothing, then we should simply unset the BackgroundFile property. and let the greeter fallback to the solid color.

Reply:
For some reason it was broken when I was writing this. Now it works fine.